File: src/main/java/com/medishield/model/Medicine.java
package com.medishield.model;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDate;

@Entity
@Table(name = "medicines")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Medicine {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(nullable = false)
    private Integer patientId;

    @Column(nullable = false, length = 150)
    private String name;

    @Column(nullable = false, length = 100)
    private String dosage;

    @Column(length = 20)
    private String frequency = "once";

    // Store as JSON string
    @Column(columnDefinition = "TEXT")
    private String timings = "[\"08:00\"]";

    @Column(columnDefinition = "TEXT")
    private String instructions = "";

    private LocalDate startDate;

    private LocalDate endDate;

    private Integer stock = 30;

    private Integer refillAt = 5;

    @Column(length = 20)
    private String color = "#2563eb";

    // Store as JSON string
    @Column(columnDefinition = "TEXT")
    private String takenDates = "[]";

    // Store as JSON string
    @Column(columnDefinition = "TEXT")
    private String missedDates = "[]";

    private Boolean isActive = true;
}

File: backend/models/Medicine.js
import { DataTypes } from 'sequelize';
import sequelize from '../config/db.js';

const Medicine = sequelize.define('Medicine', {
  id: { type: DataTypes.INTEGER, autoIncrement: true, primaryKey: true },
  patientId: { type: DataTypes.INTEGER, allowNull: false },
  name: { type: DataTypes.STRING(150), allowNull: false },
  dosage: { type: DataTypes.STRING(100), allowNull: false },
  frequency: { type: DataTypes.ENUM('once', 'twice', 'thrice', 'custom'), defaultValue: 'once' },
  timings: { type: DataTypes.JSON, defaultValue: ['08:00'] },
  instructions: { type: DataTypes.TEXT, defaultValue: '' },
  startDate: { type: DataTypes.DATEONLY },
  endDate: { type: DataTypes.DATEONLY },
  stock: { type: DataTypes.INTEGER, defaultValue: 30 },
  refillAt: { type: DataTypes.INTEGER, defaultValue: 5 },
  color: { type: DataTypes.STRING(20), defaultValue: '#2563eb' },
  takenDates: { type: DataTypes.JSON, defaultValue: [] },
  missedDates: { type: DataTypes.JSON, defaultValue: [] },
  isActive: { type: DataTypes.BOOLEAN, defaultValue: true },
}, { tableName: 'medicines', timestamps: true });

export default Medicine;
